src/apis/error.ts

```diff
@@ -42,11 +42,30 @@ export class UserError extends FiestaError {
     super(statusCode, code, message);
   }
 }
+// 리뷰 에러 클래스
+export class ReviewError extends FiestaError {
+  constructor(statusCode: number, code: string, message: string) {
+    super(statusCode, code, message);
+  }
+}
+
+function isFiestaError(error: unknown): error is FiestaError {
+  return (
+    typeof error === "object" &&
+    error !== null &&
+    "code" in error &&
+    "message" in error &&
+    "statusCode" in error
+  );
+}
+
+export function createFiestaError(error: unknown) {
+  if (!isFiestaError(error)) {
```

Comment:
P5

개인적으로 `isFiestaError` 함수까지 만들어야할까 라는 생각이 들긴하네요 만약 클래스 내부의 값이 변경이 된다면 또 수정을 해주어야하고.. 약간 불편할 것 같습니다.

```suggestion
  if (!(error instanceof FiestaError)) {
```

분기 처리를 한다면 다음과 같이 할 것 같습니다!

Reply:
P4

저는 사용자 타입 가드 로직이 보편적인 코드 스타일 중 하나고, 확장성 및 유지보수에도 장점이 있다 생각해서
함수 분리한 것 자체는 좋다고 봅니다~
다만 사용자 타입 가드 로직 내부 로직은 최적화를 할 수 있다고는 생각이 드는데,
사실 리팩터링 우선순위는 많이 낮을 것 같아요!